internal/acctest/acctest.go

```diff
@@ -262,40 +263,72 @@ func WithPlan(plan string) CreateTestServiceOpt {
 }
 
 // CreateTestService creates a test service that can be shared between tests.
-func CreateTestService(ctx context.Context, projectName, serviceName string, opts ...CreateTestServiceOpt) (func(), error) {
+// If the service already exists, it will be used as is and won't be deleted after the tests.
+func CreateTestService(t *testing.T, projectName, serviceName string, opts ...CreateTestServiceOpt) <-chan error {
 	client, err := GetTestGenAivenClient()
 	if err != nil {
-		return nil, err
+		require.NoError(t, err, "getting Aiven generated client failed")
 	}
 
-	req := &service.ServiceCreateIn{
-		ServiceName: serviceName,
-	}
+	// Creates cancel context to manage the goroutine lifecycle
+	ctx, cancel := context.WithCancel(t.Context())
+	t.Cleanup(cancel)
```

Comment:
You don't need to worry about it.
https://cs.opensource.google/go/go/+/refs/tags/go1.25.5:src/testing/testing.go;l=2459
https://cs.opensource.google/go/go/+/refs/tags/go1.25.5:src/testing/testing.go;l=1556